app/backend/src/couchers/models/events.py

```diff
@@ -166,19 +166,21 @@ def coordinates(self) -> tuple[float, float] | None:
         return get_coordinates(self.geom)
 
     @hybrid_property
-    def start_time(self) -> Any:
-        return self.during.lower
+    def start_time(self) -> datetime:
+        return not_none(self.during.lower)
 
-    @start_time.expression
-    def start_time(cls) -> Any:  # noqa: ARG003,D102
+    @start_time.inplace.expression
+    @classmethod
+    def _start_time(cls) -> ColumnElement[datetime]:
         return func.lower(cls.during)
 
     @hybrid_property
-    def end_time(self) -> Any:
-        return self.during.upper
+    def end_time(self) -> datetime:
+        return not_none(self.during.upper)
 
-    @end_time.expression
-    def end_time(cls) -> Any:  # noqa: ARG003,D102
+    @end_time.inplace.expression
+    @classmethod
+    def _end_time(cls) -> ColumnElement[datetime]:
         return func.upper(cls.during)
```

Comment:
what's going on here?

Reply:
It's a way to make hybrid_property intelligible to mypy (on an instance-level it returns a datetime, on a class level - a Column). See https://docs.sqlalchemy.org/en/20/orm/extensions/hybrid.html#defining-expression-behavior-distinct-from-attribute-behavior